### PR TITLE
feat: add hints to goosed

### PR DESCRIPTION
**Pull request by michaelneale**
goose cli loads `.goosehints` via the goose hints system (looking in home, current dir etc) so I think it makes sense that the GUI does the same.

### CODE REVIEW

**Comment by michaelneale** [comment]
cc @jsibbison-square if you could take a look and let me know if it seems ok from your knowledge

File: crates/goose-server/src/routes/reply.rs [hunk 1]
```diff
@@ -275,6 +273,13 @@ async fn handler(
 
     // Get a lock on the shared agent
     let agent = state.agent.clone();
+        
+    // Lock the agent to get a mutable reference so we can add the system
+    {
```
Comment (jsibbison-square) [inline crates/goose-server/src/routes/reply.rs]: I'm not familiar with this syntax but seems fine

Comment (jsibbison-square) [reply crates/goose-server/src/routes/reply.rs]: assuming you checked it works, its a bit confusing with the blocks and agents, can the initial agent be mutable?

Comment (michaelneale) [reply crates/goose-server/src/routes/reply.rs]: going to test it with the other branch selecting directories

**Comment by michaelneale** [comment]
image

yes tested it with a silly `.goosehints`

**Pull request merged**